js/client-chat.js

```diff
@@ -208,7 +208,14 @@
 		// highlight
 
 		getHighlight: function (message) {
-			var highlights = Dex.prefs('highlights') || [];
+			if (Array.isArray(Dex.prefs('highlights'))) {
```

Comment:
I would rather not deserialize this twice.

```suggestion
			var highlights = Dex.prefs('highlights') || {};
			if (Array.isArray(highlights)) {
				highlights = {global: highlights};
				// Migrate from the old highlight system
				Dex.prefs('highlights', {global: highlights});
```